### PR TITLE
Check if stream coordinator is quorum critial

### LINKED ISSUE #4890: Add rabbitmq-upgrade command for streams
Currently there is no command equivalent to `await_online_quorum_plus_one` that takes streams into account.

Also there would need to be some way of ensuring that the stream coordinator would not lose quorum.

### PR DESCRIPTION
**Pull request by mkuratczyk**
Since https://github.com/rabbitmq/rabbitmq-server/commit/d8ff9918, streams are considered when checking for a potential quorum loss. Now we also consider `rabbit_stream_coordinator`.

Fixes #4890

### CODE REVIEW

File: deps/rabbit/src/rabbit_stream_coordinator.erl [hunk 1]
```diff
@@ -2092,3 +2095,16 @@ transfer_leadership([Destination | _] = _TransferCandidates) ->
...
     end.
+
+online_coordinator_members() ->
+    lists:filter(fun erlang:is_pid/1,
+                 rabbit_misc:append_rpc_all_nodes(rabbit_nodes:all_running(),
+                                                  erlang, whereis, [rabbit_stream_coordinator])).
+
+is_quorum_critical() ->
+    NumAlive = length(online_coordinator_members()),
+    case ra:members(rabbit_stream_coordinator) of
```
Comment (kjnilsson) [inline deps/rabbit/src/rabbit_stream_coordinator.erl:2106]: Does this mean that if the local stream coordinator member isn't running we say it is ok to go ahead?

File: deps/rabbit/src/rabbit_upgrade_preparation.erl [hunk 2]
```diff
@@ -35,11 +35,12 @@ await_online_synchronised_mirrors(Timeout) ->
 do_await_safe_online_quorum(0) ->
     false;
 do_await_safe_online_quorum(IterationsLeft) ->
+    CriticalComponents = [rabbit_stream_coordinator],
     EndangeredQueues = lists:append(
                          rabbit_quorum_queue:list_with_minimum_quorum(),
                          rabbit_stream_queue:list_with_minimum_quorum()),
     case EndangeredQueues of
-        []  -> true;
+        []  -> not lists:member(true, [C:is_quorum_critical() || C <- CriticalComponents]);
```
Comment (kjnilsson) [inline deps/rabbit/src/rabbit_upgrade_preparation.erl:43]: I like this API? should it have its own module with a behaviour definition?

**Review by kjnilsson — CHANGES_REQUESTED** [review]
A couple of questions / suggestions

**Comment by michaelklishin** [comment]
@mergifyio rebase

**Comment by michaelklishin** [comment]
@mergifyio backport v3.11.x

**Pull request merged**